### DISCUSSION TITLE
USB support for Spike

### DISCUSSION THREAD

**Discussion by OutOfTheBots** [pybricks/support#606] [Ideas]
I work with RoboRave Australia and the University of Sunshine Coast to provide training for both students and teachers through government grants. At some events we run the can be many schools attending and we can sometimes have up to 100 students and robots. We have found using blue tooth to be a nightmare with that many robots all trying to talk on 2.4GHz and most of the day is spent trouble shooting blue tooth problems so we have rule that all robots need to be connected to PC via USB so our limited time can be more focused to teaching robotics.

Is there short term plans to add USB connection to Spike??

**Comment by laurensvalk** [comment]
> We have found using blue tooth to be a nightmare with that many robots all trying to talk on 2.4GHz

Was this with Bluetooth Classic (as used by the official LEGO apps) or Bluetooth Low Energy (used by Pybricks)?

**Reply by OutOfTheBots** [reply]
we use lots of different platforms and as a general rule of thumb if your going to be using Bluetooth in large numbers expect a lot of your time being spent dealing with Bluetooth problems.

We even get stuff like kids wanting to be mean to other kids and when we come back from a break and they turn the robots back on kids will purposely try to connect to other kids robots.

You image coming back in from a break and having up to 100 kids all searching and trying to connect to their robot at the same time.
